File: 2026-1/sesion-de-clase-03/dotnet/seguro-vehicular/Coberturas/SeguroDecorador.cs
using Seguros;

namespace Coberturas;

public class SeguroDecorador : ICotizable
{
    protected ICotizable seguro;

    public SeguroDecorador(ICotizable seguro)
    {
        this.seguro = seguro;
    }

    public virtual double CalcularCosto()
    {
        return this.seguro.CalcularCosto();
    }

    public virtual string Descripcion()
    {
        return this.seguro.Descripcion();
    }

    public Vehiculo Vehiculo()
    {
        return this.seguro.Vehiculo();
    }

    public override string ToString()
    {
        return $"Precio: {this.CalcularCosto():F2}, Descripción: {this.Descripcion()}";
    }
}

File: 2026-1/sesion-de-clase-04/java/seguro-vehicular/pe/edu/softseguros/coberturas/SeguroDecorador.java
package pe.edu.softseguros.coberturas;

import pe.edu.softseguros.modelo.Cotizable;
import pe.edu.softseguros.modelo.Vehiculo;

public class SeguroDecorador implements Cotizable {

    protected Cotizable seguro;

    public SeguroDecorador(Cotizable seguro) {
        this.seguro = seguro;
    }

    @Override
    public double CalcularCosto() {
        return this.seguro.CalcularCosto();
    }

    @Override
    public String Descripcion() {
        return this.seguro.Descripcion();
    }

    @Override
    public Vehiculo Vehiculo() {
        return this.seguro.Vehiculo();
    }

    @Override
    public String toString() {
        return String.format("Precio: %.2f, Descripción: %s",
                this.CalcularCosto(), this.Descripcion());
    }
}
